core/flyout_horizontal.js

```diff
@@ -245,16 +245,60 @@ Blockly.HorizontalFlyout.prototype.scrollToStart = function() {
 };
 
 /**
+ * Scroll the flyout to a position.
+ * @param {Number} pos The targeted scroll position.
+ */
+Blockly.HorizontalFlyout.prototype.scrollTo = function(pos) {
+  this.scrollTarget = pos * this.workspace_.scale;
+
+  // Make sure not to set the scroll target below the lowest point we can
+  // scroll to, i.e. the content height minus the view height
+  var contentWidth = this.workspace_.getMetrics().contentWidth;
+  var viewWidth = this.workspace_.getMetrics().viewWidth;
+  this.scrollTarget = Math.min(this.scrollTarget, contentWidth - viewWidth);
+
+  this.step();
+};
+
+/**
+ * Step the scrolling animation by scrolling a fraction of the way to
+ * a scroll target, and request the next frame if necessary.
+ */
+Blockly.HorizontalFlyout.prototype.step = function() {
+  if (!this.scrollTarget) return;
+  var scrollXPos = -this.workspace_.scrollX;
+  var diff = this.scrollTarget - scrollXPos;
+  if (Math.abs(diff) < 1) {
+    this.scrollbar_.set(this.scrollTarget);
+    return;
+  }
+  this.scrollbar_.set(scrollXPos + diff * 0.3);
+
+  // Polyfilled by goog.dom.animationFrame.polyfill
+  requestAnimationFrame(this.step.bind(this));
+};
+
+/**
  * Scroll the flyout.
  * @param {!Event} e Mouse wheel scroll event.
  * @private
  */
 Blockly.HorizontalFlyout.prototype.wheel_ = function(e) {
+  // remove scrollTarget to stop auto scrolling in step()
+  this.scrollTarget = null;
+
   var delta = e.deltaX;
 
+  // If we're scrolling more vertically than horizontally,
+  // use the vertical scroll delta instead
+  if (Math.abs(e.deltaY) > Math.abs(delta)) {
+    delta = e.deltaY;
+  }
+
   if (delta) {
-    if (goog.userAgent.GECKO) {
-      // Firefox's deltas are a tenth that of Chrome/Safari.
+    // Firefox's mouse wheel deltas are a tenth that of Chrome/Safari.
+    // DeltaMode is 1 for a mouse wheel, but not for a trackpad scroll event
+    if (goog.userAgent.GECKO && (e.deltaMode === 1)) {
```

Comment:
This also looks like a change that needs to go upstream.

Reply:
Cool- made a PR